pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/rebalance/TableRebalancer.java

```diff
@@ -126,24 +139,37 @@ public RebalanceResult rebalance(TableConfig tableConfig, Configuration rebalanc
     }
 
     // Reassign instances if necessary
-    Map<InstancePartitionsType, InstancePartitions> instancePartitionsMap = new TreeMap<>();
     boolean reassignInstances = rebalanceConfig
         .getBoolean(RebalanceConfigConstants.REASSIGN_INSTANCES, RebalanceConfigConstants.DEFAULT_REASSIGN_INSTANCES);
-    if (tableConfig.getTableType() == TableType.OFFLINE) {
-      instancePartitionsMap.put(InstancePartitionsType.OFFLINE,
-          getInstancePartitions(tableConfig, InstancePartitionsType.OFFLINE, reassignInstances, dryRun));
-    } else {
-      instancePartitionsMap.put(InstancePartitionsType.CONSUMING,
-          getInstancePartitions(tableConfig, InstancePartitionsType.CONSUMING, reassignInstances, dryRun));
-      instancePartitionsMap.put(InstancePartitionsType.COMPLETED,
-          getInstancePartitions(tableConfig, InstancePartitionsType.COMPLETED, reassignInstances, dryRun));
+    Map<InstancePartitionsType, InstancePartitions> instancePartitionsMap = new TreeMap<>();
+    try {
+      if (tableConfig.getTableType() == TableType.OFFLINE) {
+        instancePartitionsMap.put(InstancePartitionsType.OFFLINE,
+            getInstancePartitions(tableConfig, InstancePartitionsType.OFFLINE, reassignInstances, dryRun));
+      } else {
+        instancePartitionsMap.put(InstancePartitionsType.CONSUMING,
+            getInstancePartitions(tableConfig, InstancePartitionsType.CONSUMING, reassignInstances, dryRun));
+        instancePartitionsMap.put(InstancePartitionsType.COMPLETED,
+            getInstancePartitions(tableConfig, InstancePartitionsType.COMPLETED, reassignInstances, dryRun));
+      }
+    } catch (Exception e) {
+      LOGGER
+          .warn("Caught exception while fetching/calculating instance partitions for table: {}", tableNameWithType, e);
+      return new RebalanceResult(RebalanceResult.Status.FAILED,
+          "Caught exception while fetching/calculating instance partitions: " + e, null, null);
     }
 
     LOGGER.info("Calculating the target assignment for table: {}", tableNameWithType);
     SegmentAssignment segmentAssignment = SegmentAssignmentFactory.getSegmentAssignment(_helixManager, tableConfig);
     Map<String, Map<String, String>> currentAssignment = currentIdealState.getRecord().getMapFields();
-    Map<String, Map<String, String>> targetAssignment =
-        segmentAssignment.rebalanceTable(currentAssignment, instancePartitionsMap, rebalanceConfig);
+    Map<String, Map<String, String>> targetAssignment;
+    try {
+      targetAssignment = segmentAssignment.rebalanceTable(currentAssignment, instancePartitionsMap, rebalanceConfig);
+    } catch (Exception e) {
+      LOGGER.warn("Caught exception while calculating target assignment for table: {}", tableNameWithType, e);
+      return new RebalanceResult(RebalanceResult.Status.FAILED, "Caught exception while fetching ExternalView: " + e,
```

Comment:
This catch block only catches exception when rebalancing the table. Maybe this message needs modification?

Reply:
Good catch, I copy-pasted from the previous message and missed it